src/Uno.Toolkit.UI/Controls/TabBar/TabBarItem.Properties.cs

```diff
@@ -29,26 +27,61 @@ partial class TabBarItem
 			DependencyProperty.Register(nameof(Icon), typeof(IconElement), typeof(TabBarItem), new PropertyMetadata(null, OnPropertyChanged));
 		#endregion
 
+		// UNO TODO: Deprecate and remove BadgeVisibility and BadgeValue properties and use InfoBadge instead
 		#region BadgeVisibility
 		public Visibility BadgeVisibility
 		{
-			get { return (Visibility)GetValue(BadgeVisibilityProperty); }
-			set { SetValue(BadgeVisibilityProperty, value); }
+			get => InfoBadge?.Visibility ?? (Visibility)GetValue(BadgeVisibilityProperty);
+			set
+			{
+				SetValue(BadgeVisibilityProperty, value);
+
+				InfoBadge ??= new MUXC.InfoBadge();
+				InfoBadge.Visibility = value;
+			}
 		}
 
 		public static readonly DependencyProperty BadgeVisibilityProperty =
-			DependencyProperty.Register("BadgeVisibility", typeof(Visibility), typeof(TabBarItem), new PropertyMetadata(Visibility.Collapsed, OnPropertyChanged));
+			DependencyProperty.Register(nameof(BadgeVisibility), typeof(Visibility), typeof(TabBarItem), new PropertyMetadata(Visibility.Collapsed, OnPropertyChanged));
 		#endregion
 
 		#region BadgeValue
 		public string? BadgeValue
 		{
-			get { return (string)GetValue(BadgeValueProperty); }
-			set { SetValue(BadgeValueProperty, value); }
+			get => (InfoBadge as MUXC.InfoBadge)?.Value.ToString() ?? (string)GetValue(BadgeValueProperty);
+			set
+			{
+				SetValue(BadgeValueProperty, value);
+
+				InfoBadge ??= new MUXC.InfoBadge();
+
+				if (InfoBadge is MUXC.InfoBadge infoBadge)
+				{
+					if (int.TryParse(value, out int intValue))
+					{
+						infoBadge.Value = intValue;
+					}
+					else
+					{
+						infoBadge.IconSource = new MUXC.FontIconSource { Glyph = value };
+					}
+				}
+			}
 		}
 
 		public static readonly DependencyProperty BadgeValueProperty =
-			DependencyProperty.Register("BadgeValue", typeof(string), typeof(TabBarItem), new PropertyMetadata(default(string?), OnPropertyChanged));
+			DependencyProperty.Register(nameof(BadgeValue), typeof(string), typeof(TabBarItem), new PropertyMetadata(default(string?), OnPropertyChanged));
+		#endregion
+
+		#region InfoBadge
```

Comment:
@morning4coffe-dev the related documentation should be updated as well, please https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc

Reply:
Added the `InfoBadge` there, thank you @agneszitte. Should I also remove the old properties from the table or mark them obsolete somehow?

Reply:
@kazo0, @Xiaoy312 based on this previous comment: https://github.com/unoplatform/uno.toolkit.ui/pull/1142#discussion_r1623935490
![image](https://github.com/unoplatform/uno.toolkit.ui/assets/16295702/7fb78067-af37-4b8b-aa9f-567b157fa777)
Since we are not doing a breaking change, I don't think we should remove them from the doc, but only mention them as deprecated right?